### PR TITLE
Feat/album list

### PR DESCRIPTION
**Pull request by Bbowen100**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -102,6 +102,7 @@
     "react-redux": "^4.4.2",
     "react-router": "^2.4.1",
     "react-router-redux": "^4.0.0",
+    "recompose": "^0.20.0",
```
Comment (RSNara) [inline package.json:105]: Good choice in libraries.

File: src/action-creators/index.js [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
 export function userAuthenticated(user, token) {
   return {
     type: AUTH.LOGIN_USER_SUCCESS,
     payload: { user, token },
   };
 }
+
+export function addAlbumLink( albumLink ) {
+  return { type: ALBUM.ADD_ALBUM_LINK, payload: albumLink };
+}
```
Comment (RSNara) [inline src/action-creators/index.js:12]: Most excellent action creator.

File: src/components/link-input/index.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import * as recompose from 'recompose';
+
+import React from 'react';
```
Comment (RSNara) [inline src/components/link-input/index.js:2]: I usually tend to group the imports together. But no biggy.

File: src/containers/about-page.js [hunk 4]
```diff
@@ -20,7 +20,6 @@ export default connect(
 )(AboutPage);
 
 function mapStateToProps() {
-  return {};
 }
```
Comment (RSNara) [inline src/containers/about-page.js:23]: Hm. We should probably leave the empty object there. It's less obvious what returning undefined from `mapStateToProps` does.

File: src/action-creators/index.js (hunk 2 above)
Comment (mandishshah) [inline src/action-creators/index.js:12]: Payload should be an object and avoid all of Raman's comments :)

File: src/constants/index.js [hunk 5]
```diff
@@ -6,6 +6,10 @@ export const AUTH = createActions('App', [
   'GET_USER_INFO',
 ]);
 
+export const ALBUM = createActions('ALBUM', [
```
Comment (mandishshah) [inline src/constants/index.js]: Shouldn't this be Albums_Add_Album_Links? Just wondering

File: src/containers/about-page.js (hunk 4 above)
Comment (mandishshah) [inline src/containers/about-page.js:23]: Listen to Raman on this one

File: src/index.js [hunk 6]
```diff
@@ -17,6 +17,7 @@ import './styles/index.css';
 const store = configureStore({});
 const history = syncHistoryWithStore(browserHistory, store);
 store.runSaga(rootSaga);
+window.store = store;
```
Comment (mandishshah) [inline src/index.js]: Are you good bruh? Remove this shit now!

File: src/reducers/album-links.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+const INITIAL_STATE = I.Map({
+  albumLinks: I.List(),
+});
+
+const albumlinks = (state = INITIAL_STATE, action) => {
+  switch (action.type) {
+    case ALBUM.ADD_ALBUM_LINK:
+      const link = action.payload;
+      return state.update('albumLinks', (albumLinks) => albumLinks.push(link));
```
Comment (mandishshah) [inline src/reducers/album-links.js:12]: We should make sure if the album link already exist or not

File: src/store/routes.js [hunk 8]
```diff
@@ -4,13 +4,15 @@ import { push } from 'react-router-redux';
...
 import * as AC from '../action-creators';
+import AlbumLinksPage from '../containers/album-links-page';
 
 export default (store) => (
   <Router history={browserHistory}>
     <Route path="/" component={ App }>
       <IndexRoute to="/about"/>
       <Route path="about" component={ AboutPage }/>
       <Route path="auth/facebook" onEnter={() => checkAuth(store)}/>
+      <Route path="album-links" component={ AlbumLinksPage }/>
```
Comment (mandishshah) [inline src/store/routes.js]: Just user albums?

**Comment by mandishshah** [comment]
Good luck rebasing :)

File: src/containers/about-page.js (hunk 4 above)
Comment (Bbowen100) [inline src/containers/about-page.js:23]: this line got replaced with this:
return { albumLinks: Selectors.getAlbumLinks(state) };

**Pull request closed without merging**